flink-end-to-end-tests/flink-taskmanager-direct-memory-test/src/main/java/org/apache/flink/streaming/tests/TaskManagerDirectMemoryTestProgram.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.tests;
+
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
+import org.apache.flink.streaming.api.functions.source.RichParallelSourceFunction;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * Test program for taskmanager direct memory consumption.
+ */
+public class TaskManagerDirectMemoryTestProgram {
+	private static final ConfigOption<Integer> RUNNING_TIME_IN_SECONDS = ConfigOptions
+		.key("test.running_time_in_seconds")
+		.defaultValue(120)
+		.withDescription("The time to run.");
+
+	private static final ConfigOption<Integer> RECORD_LENGTH = ConfigOptions
+		.key("test.record_length")
+		.defaultValue(2048)
+		.withDescription("The length of record.");
+
+	private static final ConfigOption<Integer> MAP_PARALLELISM = ConfigOptions
+		.key("test.map_parallelism")
+		.defaultValue(1)
+		.withDescription("The number of map tasks.");
+
+	private static final ConfigOption<Integer> REDUCE_PARALLELISM = ConfigOptions
+		.key("test.reduce_parallelism")
+		.defaultValue(1)
+		.withDescription("The number of reduce tasks.");
+
+	public static void main(String[] args) throws Exception {
+		// parse the parameters
+		final ParameterTool params = ParameterTool.fromArgs(args);
+
+		final int runningTimeInSeconds = params.getInt(RUNNING_TIME_IN_SECONDS.key(), RUNNING_TIME_IN_SECONDS.defaultValue());
+		final int recordLength = params.getInt(RECORD_LENGTH.key(), RECORD_LENGTH.defaultValue());
+		final int mapParallelism = params.getInt(MAP_PARALLELISM.key(), MAP_PARALLELISM.defaultValue());
+		final int reduceParallelism = params.getInt(REDUCE_PARALLELISM.key(), REDUCE_PARALLELISM.defaultValue());
```

Comment:
It might not make sense to adjust these parameters by any values, otherwise we can mock the specific scenario to exceed the netty memory overhead. Better to make them static in program.

Reply:
Change the `recordLength` variable into constant